src/Uno.UI/Helpers/ElevationHelper.cs

```diff
@@ -0,0 +1,102 @@
+﻿#if !NETFX_CORE
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Reflection;
+using Windows.UI;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Shapes;
+using Uno.Extensions;
+using Uno.Logging;
+using Uno.UI.Helpers;
+
+#if __IOS__ || __MACOS__
+using CoreGraphics;
+#endif
+
+namespace Uno.UI.Helpers
+{
+	internal static class ElevationHelper
+	{
+#if __IOS__ || __MACOS__
+		internal static void SetElevation(this DependencyObject element, double elevation, Color shadowColor, CGPath path = null)
+#else
+		internal static void SetElevation(this DependencyObject element, double elevation, Color shadowColor)
+#endif
+		{
+#if __ANDROID__
+			if (element is Android.Views.View view)
+			{
+				AndroidX.Core.View.ViewCompat.SetElevation(view, (float)Uno.UI.ViewHelper.LogicalToPhysicalPixels(elevation));
+				if(Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.P)
+				{
+					view.SetOutlineAmbientShadowColor(shadowColor);
+					view.SetOutlineSpotShadowColor(shadowColor);
+				}
+			}
+#elif __IOS__ || __MACOS__
+#if __MACOS__
+			if (element is AppKit.NSView view)
+#else
+			if (element is UIKit.UIView view)
+#endif
```

Comment:
I think the code in this class is a copy from the "toolkit".  Maybe we should delete the original to prevent duplicating code?

Reply:
Yes, I actually moved the code here - toolkit now calls this `ElevationHelper` 🙂 (except for UWP, which stayed in toolkit)